app/models/application_letter.rb

```diff
@@ -20,7 +20,23 @@ class ApplicationLetter < ActiveRecord::Base
   validates :grade, presence: true, numericality: { only_integer: true }
   validates :vegeterian, :vegan, :allergic, inclusion: { in: [true, false] }
   validates :vegeterian, :vegan, :allergic, exclusion: { in: [nil] }
+  validate :deadline_cannot_be_in_the_past, :if => Proc.new { |letter| !(letter.status_changed?) }
```

Comment:
What is `Proc` and why is it needed? 
It isn't clear to me why you used it (after reading [the docs](https://ruby-doc.org/core-2.2.0/Proc.html).